prefq/server.py

```diff
@@ -255,14 +290,39 @@ def main():
         default=DEFAULT_DEBUG,
         help="Specify debug mode (default: False)",
     )
+    parser.add_argument(
+        "--sshkey",
+        type=str,
+        default=None,
+        help="Specify SSH key filepath (default: None)",
+    )
+    # pylint: disable=R0801
+    parser.add_argument(
+        "--pw",
+        type=str,
+        default=None,
+        help="Specify Password for request authentication (default: None)",
+    )
 
     args = parser.parse_args()
 
     host = args.host
     port = args.port
     debug = args.debug
+    sshkey = args.sshkey
+    server_pw = args.pw
+
+    is_authentication_desired = (server_pw is not None) and (sshkey is not None)
```

Comment:
Why do we need both an ssh key and a password? Shouldn't the key suffice?

Reply:
using the challenge-request approach not really, in the updated version I've removed the password